packages/amplify_core/lib/src/types/notifications/push/pinpoint_event_type.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/// {@template amplify_core.push.pinpoint_event_type}
+/// Enum that indicates the different types recorded with Pinpoint
+/// {@endtemplate}
+enum PinpointEventType {
+  notificationOpened,
+  backgroundMessageReceived,
+  foregroundMessageReceived,
+}
+
+extension PinpointEventTypeExtension on PinpointEventType {
```

Comment:
nit: could use enhanced enum instead of extension as we are enforcing 2.18+ Dart SDK, same to other enum.

```dart
enum PinpointEventType {
  notificationOpened('opened_notification'),
  backgroundMessageReceived('received_background'),
  foregroundMessageReceived('received_foreground');
  
  const PinpointEventType(this.name);
  
  final String name;
}
```